agent/proxycfg/snapshot.go

```diff
@@ -532,6 +539,15 @@ func (s *ConfigSnapshot) Leaf() *structs.IssuedCert {
 	}
 }
 
+// RootPEMs returns all PEM-encoded public certificates for the root CA.
+func (s *ConfigSnapshot) RootPEMs() string {
+	var rootPEMs string
+	for _, root := range s.Roots.Roots {
+		rootPEMs += ca.EnsureTrailingNewline(root.RootCert)
+	}
+	return rootPEMs
```

Comment:
nit: could use `strings.Builder` here to save memory in case there's a non-trivial number of roots